### ISSUE TITLE
[Accessibility] Narrator should announce "end of line'" instead of "Space" when focusing on last text position in comboBox and textBox

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#4749]
* .NET Core Version: 
.Net 6.0 SDK build: 6.0.100-Preview.4.21180.17

* Have you experienced this same bug with .NET Framework?: 
 No
* More Info
This is a regression issue, found after the fix of PR: https://github.com/dotnet/winforms/pull/4540

**Repro Steps:**
1. Open attached application in VS, build and run it
[WinFormsApp11.zip](https://github.com/dotnet/winforms/files/6235603/WinFormsApp11.zip)
2. Open Narrator to focus the last position of item in comboBox and TextBox
 for comboBox, use mouse expand dropdown list, choose any of item, then move mouse to last position in edit part
for textBox, from other control tab to textBox, then move mouse to last position in edit part
**Actual behavior:**
Narrator announces "end of line" when moving mouse to last position in edit of comboBox and textBox
![image](https://user-images.githubusercontent.com/26474449/113126405-2de2b700-924a-11eb-9fad-8a4634e051ce.png)
![spaceIssue](https://user-images.githubusercontent.com/26474449/113126867-b2353a00-924a-11eb-8fe7-8d1e2efbd4dd.gif)

**Expected behavior:**
Narrator should announce "end of line" when moving mouse to last position in edit of comboBox and textBox

**Comment by SergeySmirnov-Akvelon** [comment]
This regression is the result of adding text pattern support (#2701) to the `TextBox`. This issue is reproducible for the `ComboBox`, because we reused the same logic to support the text pattern (#4540)
